scripts/gencode.py

```diff
@@ -185,6 +185,9 @@ def gen_init():
     args = "void" if cmd is None else formatArgs(cmd['outputs'], cmd['inputs'])
 
     print "result_t %s_init(%s) {" % (c_ifacename, args)
+    print "\tif(%s_initializations++ > 0) {" % c_ifacename
+    print "\t\treturn RESULT_OK;"
+    print "\t}"
     print "\tresult_t res;"
     print "\tres = sm_get_service(&%s_object, \"%s\");" % (c_ifacename, ifacename)

```

Comment:
IPC modules should initialize `sm` before calling `sm_get_service` so as to hide their dependency on it, and make sure to also close it afterwards. You also need to decrease `initializations` on failure. See https://github.com/reswitched/libtransistor/blob/master/lib/ipc/vi.c#L42
I normally don't nitpick on this, but I'd like autogenerated code to follow our [coding style](https://github.com/reswitched/libtransistor/blob/master/STYLE.md), so it should look more like this:

```c
result_t %s_init(%s) {
    if(%s_initializations++ > 0) {
        return RESULT_OK;
    }

    result_t r;
    r = sm_init();
    if(r) {
        goto fail;
    }

    r = sm_get_service(&%s_object, "%s");
    if(r) {
        goto fail_sm;
    }

    sm_finalize();
    return RESULT_OK;

fail_sm:
    sm_finalize();
fail:
    %s_initializations--;
    return r;
}
````